src/bin/pg_autoctl/pgctl.c

```diff
@@ -114,6 +114,54 @@ pg_ctl_version(PostgresSetup *pgSetup)
 }
 
 
+/*
+ * set_pg_ctl_from_PG_CONFIG sets given pgSetup->pg_ctl to the pg_ctl binary
+ * installed in the bindir of the target Postgres installation:
+ *
+ *  $(${PG_CONFIG} --bindir)/pg_ctl
+ */
+bool
+set_pg_ctl_from_config_bindir(PostgresSetup *pgSetup, const char *pg_config)
+{
+	char pg_ctl[MAXPGPATH] = { 0 };
+	Program prog = run_program(pg_config, "--bindir", NULL);
+
+	char *bindir;
+	char *lines[1];
+
+	if (prog.returnCode != 0)
+	{
+		errno = prog.error;
+		log_error("Failed to run \"pg_config --bindir\" using program \"%s\": %m",
+				  pg_config);
+		free_program(&prog);
+		return false;
+	}
+
+	if (splitLines(prog.stdOut, lines, 1) != 1)
+	{
+		log_error("Unable to parse output from pg_config --bindir");
+		free_program(&prog);
```

Comment:
It seems `free_program` in the later function exit points is missing (i.e. the `return true` and the other `return false`.

Reply:
Thanks, fixed now.